### PR TITLE
machine/stm32f4disco: add target file for updated board

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds an additional target file for updated versions of the STM32F4 Discovery boards that have the updated ST-Link-2-1 onboard.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -259,6 +259,10 @@ smoketest:
 	@$(MD5SUM) test.hex
 	$(TINYGO) build -size short -o test.hex -target=stm32f4disco        examples/blinky2
 	@$(MD5SUM) test.hex
+	$(TINYGO) build -size short -o test.hex -target=stm32f4disco-1       examples/blinky1
+	@$(MD5SUM) test.hex
+	$(TINYGO) build -size short -o test.hex -target=stm32f4disco-1       examples/blinky2
+	@$(MD5SUM) test.hex
```
Comment (aykevl) [inline Makefile]: Only one smoke test should be fine. The only thing that an extra smoke test tests is that the `LEDx` constants exist, which is not something that is likely to break unintentionally (and it slows down tests).

**Review by aykevl — COMMENTED** [review]
That's unfortunate. There is no interface that will work for both (`stlink-v2-1` or `stlink-v2`)?

**Comment by deadprogram** [comment]
I was not able to get openocd to recognize the onboard jlink adaptor unless I had the full name with the `-1` suffix.

File: Makefile (hunk 1 above)
Comment (deadprogram) [reply Makefile]: Removed extra smoketest.

**Comment by aykevl** [comment]
That is unfortunate. And the reverse doesn't work either, driving the older boards with `stlink-v2-1`?

**Comment by deadprogram** [comment]
I do not have the older one to test with, so not sure on that.

**Pull request merged**